superx/routes/home.py

```diff
@@ -76,27 +80,49 @@ def cart():
 
 def livesearch():
     """
-    returns search functin using jquery data and ajax so not to redirect
+    returns search function using jquery data and ajax so not to redirect
     """
 
-    json_list_of_items = []
+    products = []
+    search_res = request.form.get("input").strip()
+    branches_code_list = session['branches_data']
 
     # if search_res is empty string, return empty json
-    search_res = request.form.get("input").strip()
     if not search_res:
-        return render_template('products_table.html', products=json_list_of_items)
+        return render_template('products_table.html', products=products)
 
-    products_list = db.session.query(Product).order_by(Product.name)\
+    # query all products from DB that contains search_res in their name
+    products_list = db.session.query(Product).order_by(Product.name) \
         .filter(Product.name.contains(search_res)).all()
 
-    for item in products_list:
-        json_list_of_items.append({
-            "id": item.id,
-            "name": item.name,
-            "quantity": float(item.quantity),
-            "unit_of_measure": item.unit_of_measure
-        })
-    return render_template('products_table.html', products=json_list_of_items)
+    for item_count, item in enumerate(products_list):
+
+        # after NUMBER_OF_ITEMS_TO_SHOW items added break from the loop
+        if item_count == NUMBER_OF_ITEMS_TO_SHOW:
```

Comment:
As part of your query, you can limit the number of results it returns and use offset to get the next set of records.
You can also you paginate to handle that for you.
https://topic.alibabacloud.com/a/several-pagination-methods-of-sqlalchemy-and-flask-sqlalchemy_8_8_30013584.html

Reply:
I think that in our case we need to import all the BranchPrice objects that match the query because we don't know which object belongs to which Branch until the next step in the code. So if we query a limited number of entries in the table we might skip the correct ones.

Reply:
You may want to look at how you can combine all 3 of those queries into one query so you are only asking the database for the data that you want.